### DISCUSSION TITLE
Proxy

### DISCUSSION THREAD

**Discussion by frizib** [PrismarineJS/bedrock-protocol#246] [Q&A] [answered]
### How to cancel a message ' .help' to the general chat?
![image](https://user-images.githubusercontent.com/28644006/180632055-3c6206d1-c003-4364-be8c-3fe8430143db.png)

How do I make an 'invisible dialog for everyone' not on my own behalf?
![image](https://user-images.githubusercontent.com/28644006/180632152-056028cd-5487-40d7-8c59-2ae020c7b3da.png)

**Comment by extremeheat** [comment]
In relay : When you handle a server bound packet event, the second argument is an object. You can set the .cancelled property to true to cancel the packet. 


.on("serverbound", (packet, data) => data.cancelled=true))

**Comment by frizib** [comment]
Somehow it doesn't work, I don't understand
[*LINK*](https://pastebin.com/sQL5EcvH)

**Comment by extremeheat** [comment]
Your code is bad here

```
Server failed to read packet ReferenceError: name is not defined
    at Relay.<anonymous> (/root/bot/bot.js:33:1)        <---
```

**Comment by frizib** [comment]
I wrote as I was told. Is not it so?

**Comment by frizib** [comment]
I sat down, found a way to remove the nickname near the list of commands.

Now you need to understand how to properly cancel sending text to the global chat when calling the command menu.

**Comment by extremeheat** [comment]
Please elaborate on the problem and the expected outcome. I'm not sure what you're asking. 

Are you asking for an example to block a clientbound packet from the server ?

**Comment by frizib** [comment]
I need to make it so that when a user on the server writes any command (which is in the proxy), it is not sent to the general chat.

**Comment by extremeheat** [comment, marked as answer]
Please post questions on the discord and leave issue tracker for bugs.

```js
  // Client is sending a message to the server
  player.on('serverbound', ({ name, params }, options) => {
    if (name === 'text') { // Intercept chat message
      options.canceled = true // will no longer be sent to server
    }
  })
```

**Comment by frizib** [comment]
> Please post questions on the discord and leave issue tracker for bugs.
> 
> ```js
>   // Client is sending a message to the server
>   player.on('serverbound', ({ name, params }, options) => {
>     if (name === 'text') { // Intercept chat message
>       options.cancelled = true // will no longer be sent to server
>     }
>   })
> ```

It doesn't work

**Reply by extremeheat** [reply]
My mistake. It's spelled "canceled".

**Reply by frizib** [reply]
> My mistake. It's spelled "canceled".

thanks ❤

**Comment by extremeheat** [comment]
> Please post questions on the discord and leave issue tracker for bugs.

Will move this to a discussion as this is not an issue with bedrock-protocol.
